docs/testplan/server-side-copy-testplan.md

```diff
@@ -0,0 +1,53 @@
+|                        | Copy 10000 Or More Parts                                   | If a user uploads an object with 10000 parts or more and server side copies it from the source object to the destination object, it should be possible                                                                                                               |          |
+|                        | Copy 10000 Or More Segments                                   | If a user uploads an object with 10000 segments or more and server side copies it from the source object to the destination object, it should be possible                                                                                |          |
+|                        | Copy Objects More than 5GB                                    | If objects are more than 5GB in size, then upload-part-copy api must be used for server side copy                                                                                                                                                       |          |
+|                        | Copy Objects Less than 5GB                                    | If objects are less than 5GB in size, then server side copy must be an atomic operation                                                                                                                                                                                 |          |
+|                        | Copy 0 bytes object                                           | User should copy 0 bytes object without any error                                                                                                                                                                                                                       |          |
+|                        | Copy inline segment                                           | User should copy inline segment without any errors                                                                                                                                                                                                                      |          |
+|                        | Copy remote segment                                           | User should copy inline segment without any errors                                                                                                                                                                                                                      |          |
+|                        | Start server-side copying process and drop connection         | File should be copied even when connection is dropped                                                                                                                                                                                                                   |          |
```

Comment:
I'm confused "drop connection". There will be first a call to BeginCopyObject satellite endpoint, and then a call to FinishCopyObject. If the connection is dropped before FinishCopyObject is called, the object won't be copied.

Maybe the confusion comes from me using the term "atomic operation". It was more from the satellite perspective, in the sense that we should not partially copied objects in the db.

Reply:
Updated